sys/net/link_layer/nomac/nomac.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+#include <errno.h>
+#include <string.h>
+
+#include "netapi.h"
+#include "msg.h"
+#include "netdev/base.h"
+#include "thread.h"
+
+#include "nomac.h"
+
+#ifdef MODULE_NETDEV_DUMMY
+#include "netdev_dummy.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define NOMAC_MSG_QUEUE_SIZE    (16)
+
+static struct {
+    kernel_pid_t registrar_pid; /**< Thread recipient is registered to */
```

Comment:
If we put the registry into the thread's stack this shouldn't be to much of a problem. There was however a specific reason, why I did not do that… can't remember now… Maybe tomorrow, when I address @haukepetersen's comments I can recall it.